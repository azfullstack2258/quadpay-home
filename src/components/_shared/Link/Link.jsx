import { Link as RouterLink } from 'react-router-dom';
import styled from 'styled-components';

export const P = styled.p.attrs((props) => ({
  className: `text-${props.color || 'primary'}`,
}))``;

export const Link = styled(P).attrs((props) => ({
  as: RouterLink,
  to: props.to,
  className: `underline hover:text-secondary hover:cursor-pointer text-sm`,
}))``;

export default Link;
