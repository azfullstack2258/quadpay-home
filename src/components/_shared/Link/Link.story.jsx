import React from 'react';
import { storiesOf } from '@storybook/react';

import Example from '@/utils/storybook-wrapper';
import Stack from '@/utils/storybook';

import Link from './Link';

storiesOf('Components/Link', module).add('appearance', () => (
  <Example title="Link appearance">
    <Stack>
      <Link to="#" onClick={() => console.log('Link clicked')}>
        Link
      </Link>
    </Stack>
  </Example>
));
