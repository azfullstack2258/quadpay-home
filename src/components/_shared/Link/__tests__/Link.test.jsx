import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Link from '../Link';

describe('<Link />', () => {
  it('should render a prop', () => {
    const { container, getByText, findByTestId } = render(
      <Router>
        <Link to="#">Link</Link>
      </Router>
    );

    expect(container).toMatchSnapshot();

    const linkElement = getByText(/link/i);
    expect(linkElement).toBeInTheDocument();
  });
});
