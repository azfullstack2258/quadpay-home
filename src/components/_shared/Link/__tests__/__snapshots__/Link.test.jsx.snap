// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render a prop 1`] = `
<div>
  <a
    class="sc-AxjAm sc-AxirZ dqVvwz text-primary underline hover:text-secondary hover:cursor-pointer text-sm"
    href="/"
  >
    Link
  </a>
</div>
`;
